### DISCUSSION TITLE
Accuracy of identifying pdf table

### DISCUSSION THREAD

**Discussion by hhhhjjj** [jsvine/pdfplumber#629] [Q&A] [answered]
Do you have the accuracy of identifying pdf table？

**Comment by jsvine** [comment, marked as answer]
Hi @hhhhjjj, and thanks for your interest in this library. I take your question to mean something like: Has `pdfplumber`'s table-detection algorithm been tested against a benchmark, and evaluated re. whether the tables are extracted correctly? (If that's not your question, please do let me know.)

The short answer is: no, it has not. The longer answer is that `pdfplumber`'s table-detection algorithm does not take a probabilistic approach, but rather a deterministic one. And although it aims to provide utility with just its default settings, it provides the most utility when you customize the detection settings to the particular PDF you are parsing. So although it's theoretically possible to test `pdfplumber`'s default table-detection approach against an external benchmark, and it might provide some helpful comparisons for people who want to bulk-and-auto-extract tables from a large number of different PDFs at once, I don't think would represent `pdfplumber`'s typical usage.

**Reply by hhhhjjj** [reply]
Thank you for your answer
